.github/workflows/ci.yml

```diff
@@ -19,6 +19,33 @@ jobs:
       - name: Run tests
         run: bundle exec rake test:units
 
+  test-legacy:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        ruby: ["2.0", "2.1", "2.2"]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+      - name: Run tests
+        run: bundle exec rake test:units
+
+  test-all:
```

Comment:
🗒️ This aggregates all of the `test*` the matrix runs into a single check. That way the branch protection rule becomes much simpler.